searx/settings_loader.py

```diff
@@ -30,29 +31,32 @@ def load_yaml(file_name):
 
 
 def get_default_settings_path():
-    return check_settings_yml(join(searx_dir, 'settings.yml'))
+    return existing_filename_or_none(join(searx_dir, 'settings.yml'))
 
 
-def get_user_settings_path():
-    # find location of settings.yml
+def get_user_settings_path() -> Optional[str]:
+    """Get an user settings file.
+
+    By descending priority:
+    * environ['SEARXNG_SETTINGS_PATH']
+    * '/etc/searxng/settings.yml'
+    * None
+
+    Note: check only SEARXNG_SETTINGS_PATH if environ['SEARXNG_SETTINGS_PATH'] is 1 or true
+    """
+
+    # check the environment variable SEARXNG_SETTINGS_PATH
+    # if the environment variable is defined, this is the last check
     if 'SEARXNG_SETTINGS_PATH' in environ:
-        # if possible set path to settings using the
-        # enviroment variable SEARXNG_SETTINGS_PATH
-        return check_settings_yml(environ['SEARXNG_SETTINGS_PATH'])
+        return existing_filename_or_none(environ['SEARXNG_SETTINGS_PATH'])
 
+    # if SEARXNG_DISABLE_ETC_SETTINGS don't look any futher
     if environ.get('SEARXNG_DISABLE_ETC_SETTINGS', '').lower() in ('1', 'true'):
```

Comment:
Should we document `SEARXNG_DISABLE_ETC_SETTINGS` [here](https://docs.searxng.org/admin/engines/settings.html?#settings-yml-location) .. and that `SEARXNG_SETTINGS_PATH` takes precedence?

IDK what is better: should `SEARXNG_DISABLE_ETC_SETTINGS` take precedence over `SEARXNG_SETTINGS_PATH` / or with other words: should `SEARXNG_DISABLE_ETC_SETTINGS` disable all kind of custom settings?

Reply:
This is an internal variable for the tests, so `/etc/searxng/settings.yml` does not break the tests:
* https://github.com/searxng/searxng/blob/8bdc6986a1488c6ef72b6ab3512d570f5868f4a7/tests/__init__.py#L7
* https://github.com/searxng/searxng/blob/8bdc6986a1488c6ef72b6ab3512d570f5868f4a7/tests/unit/test_settings_loader.py#L56-L122

==> 
* there is no need to document this.
* `SEARXNG_DISABLE_ETC_SETTINGS` must not skip `SEARXNG_SETTINGS_PATH` : the tests check `SEARXNG_SETTINGS_PATH` is working as expected.